src/database/index.tsx

```diff
@@ -368,28 +373,52 @@ export const saveFacility = async (facility: any): Promise<void> => {
       // Ensures we don't store any attributres that our model does not
       // know about.
       facility.modelInputs = pick(facility.modelInputs, persistedKeys);
+      facility.modelInputs.updatedAt = currrentTimestamp();
+      // TODO: For now, this assumes we're always entering data as of "today"
+      // However, in the near future, we should allow for a user submitted
+      // observed at value.  If it is not provided default to today. For dates
+      // observed in the past we'll have to assume the time portion of the
+      // timestamp is startOfDay** since we won't otherwise have any precision
+      // information.
+      //
+      // ** https://date-fns.org/v1.29.0/docs/startOfDay
+      facility.modelInputs.observedAt = new Date();
     }
 
+    const db = await getDb();
+    const batch = db.batch();
```

Comment:
`batch` allows us to do multiple writes (i.e. saving the model and its version) atomically